### DISCUSSION TITLE
How to get around the records return hard limit of 500?

### DISCUSSION THREAD

**Discussion by galatea99** [pocketbase/pocketbase#3897] [Q&A] [answered]
When using the API to retrieve records, I found that even with perPage=9999, the max number of returned records is always capped at 500. e.g. /api/collections/somecollection/records?sort=-date_in&perPage=9999. will only return 500 records. 
Is this a bug or a feature? How to get around it without calling it multiple times to get number of pages and then loop through the pages?

**Comment by ganigeorgiev** [comment, marked as answer]
The max 500 records limit is an intended behavior.

There were requests to increase it in the past but I'm still not sure about it.
Even if you have a very compact collection structure, fetching thousands of records at once via a web API could have significant performance impact because we'll have to keep the records in memory at least for the duration of the response serialization (and it could be even worst when combined with something like `expand`).

It exists in my todo and eventually may get added as collection option in the future but that will be most likely together with https://github.com/pocketbase/pocketbase/issues/1232.

For now the recommended approach to fetch more than 500 records at once is to use the SDKs `getFullList()` helper (_under the hood it will send `getList` requests on batches until all items are fetched_):

```js
const records = await pb.collection("example").getFullList();
```

**Reply by galatea99** [reply]
So the short answer is no for now. Thank you for quick reply.
